### DISCUSSION TITLE
Step with Header navigation, the status default is done, how can I change that?

### DISCUSSION THREAD

**Discussion by lpnueg4** [zauberzeug/nicegui#2078] [Q&A]
### Question

~How can I set Step "done" and "error" status?~

[https://quasar.dev/vue-components/stepper#header-navigation](https://quasar.dev/vue-components/stepper#header-navigation)

in Nicegui, when I use "**Header navigation**", the before Step default is "done" state.

![image](https://github.com/zauberzeug/nicegui/assets/16138783/7999a63d-5351-464e-af6c-80114d803098)

in quasar, use "Header navigation", the before Step default is "not done" state.
![image](https://github.com/zauberzeug/nicegui/assets/16138783/c1d9ff41-b29e-4df8-bdf2-131c0fbca0de)

Is there any way to achieve this?

```python
from nicegui import ui


def oops():
    ingredients.props("error")
    stepper.next()


with ui.stepper().props("header-nav").classes("w-full") as stepper:
    with ui.step("Preheat"):
        ui.label("Preheat the oven to 350 degrees")
        with ui.stepper_navigation():
            ui.button("Next", on_click=stepper.next)
    with ui.step("Ingredients") as ingredients:
        ui.label("Mix the ingredients")
        with ui.stepper_navigation():
            ui.button("Next", on_click=oops)
            ui.button("Back", on_click=stepper.previous).props("flat")
    with ui.step("Bake"):
        ui.label("Bake for 20 minutes")
        with ui.stepper_navigation():
            ui.button("Done", on_click=lambda: ui.notify("Yay!", type="positive"))
            ui.button("Back", on_click=stepper.previous).props("flat")

ui.run()
```

When I use navigation, I want keep before steps status is "not done"

**Comment by natankeddem** [comment]
I altered the documentation example:
```py
from nicegui import ui

def oops():
    ingredients.props("error")
    stepper.next()

with ui.stepper().props("vertical").classes("w-full") as stepper:
    with ui.step("Preheat"):
        ui.label("Preheat the oven to 350 degrees")
        with ui.stepper_navigation():
            ui.button("Next", on_click=stepper.next)
    with ui.step("Ingredients") as ingredients:
        ui.label("Mix the ingredients")
        with ui.stepper_navigation():
            ui.button("Next", on_click=oops)
            ui.button("Back", on_click=stepper.previous).props("flat")
    with ui.step("Bake"):
        ui.label("Bake for 20 minutes")
        with ui.stepper_navigation():
            ui.button("Done", on_click=lambda: ui.notify("Yay!", type="positive"))
            ui.button("Back", on_click=stepper.previous).props("flat")

ui.run()
```

**Reply by lpnueg4** [reply]
@natankeddem I want set props ":done=false", but it not work.

**Reply by natankeddem** [reply]
Why not let the `next`/`previous` methods control the `done`? What are you trying to accomplish here? The step still shows as error in my previous example.  If you don't want to allow the user to complete the stepper just make it so the "Done" button is disabled or not visible if the steps haven't been completed. You need to better explain your problem/use case here.

Also, try not to change the original post, It makes the discussion difficult to follow...

**Comment by falkoschindler** [comment]
NiceGUI's stepper automatically sets steps before the current one to "done". You can circumvent this logic by deriving a custom stepper class like this:

```py
class MyStepper(ui.stepper):

    def _handle_value_change(self, value) -> None:
        super()._handle_value_change(value)
        for step in self:
            step._props[':done'] = step._props['name'] == self.value

with MyStepper().props('header-nav').classes('w-full') as stepper:
    ...
```

**Reply by lpnueg4** [reply]
@falkoschindler Cool，thanks for the example.

**Reply by lpnueg4** [reply]
@natankeddem You don't understand what I mean. I want to use navigation to switch steps, but some steps have not been operated yet, so it status should not be done.
